src/edu/csus/ecs/pc2/ui/SubmitSampleRunsPane.java

```diff
@@ -0,0 +1,877 @@
+		leftSide.setPreferredSize(new Dimension(230, 230));
+		leftSide.setVerticalScrollBarPolicy(ScrollPaneConstants.VERTICAL_SCROLLBAR_AS_NEEDED);
+		leftSide.setWheelScrollingEnabled(true);
+		splitPane.setLeftComponent(leftSide);
+
+		// centerPane2.add(LogPanel, BorderLayout.SOUTH);
+		JScrollPane logScrollPanel = new JScrollPane(LogPanel);
+		logScrollPanel.setPreferredSize(new Dimension(60, 60));
+		logScrollPanel.setVerticalScrollBarPolicy(ScrollPaneConstants.VERTICAL_SCROLLBAR_AS_NEEDED);
+		logScrollPanel.setWheelScrollingEnabled(true);
+		splitPane.setRightComponent(logScrollPanel);
+		
+		
+		selectedNosLabel.setToolTipText("None selected");
+		selectedNosLabel.addMouseListener(new MouseAdapter() {
+			@Override
+			public void mouseClicked(MouseEvent e) {
+			    showJudgementTypeList();
+			}
+		});
+		submitNoSamplesPane.add(selectedNosLabel);
+		JButton selectJudementTypesButton = new JButton("Select");
+		selectJudementTypesButton.setToolTipText("Select Judgement Types");
+		selectJudementTypesButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+			    showJudgementTypeList();
+			}
+		});
+		
+		submitNoSamplesPane.add(selectJudementTypesButton);
+		
+		JLabel lblNewLabel_2 = getWhatsThisOne();
+		submitNoSamplesPane.add(lblNewLabel_2);
+		
+		JPanel submitNoSamplesPane_1 = new JPanel();
+		FlowLayout flowLayout_2 = (FlowLayout) submitNoSamplesPane_1.getLayout();
+		flowLayout_2.setAlignment(FlowLayout.LEFT);
+		submitNoSamplesPane_1.setBorder(new TitledBorder(null, "", TitledBorder.LEADING, TitledBorder.TOP, null, null));
+		submitNoSamplesPane_1.setBounds(33, 161, 650, 32);
+		controlsPane.add(submitNoSamplesPane_1);
+		   
+        checkBoxProblems = new JCheckBox("Problems");
+        checkBoxProblems.setVerticalAlignment(SwingConstants.TOP);
+        checkBoxProblems.setToolTipText("Select Problems");
+        checkBoxProblems.setSelected(true);
+        submitNoSamplesPane_1.add(checkBoxProblems);
+        
+        submitAllProblemsRadioButton = new JRadioButton("Submit All");
+        submitAllProblemsRadioButton.setSelected(true);
+        submitNoSamplesPane_1.add(submitAllProblemsRadioButton);
+        
+        submitSelectedProblemsRadioButton = new JRadioButton("Submit Selected");
+        submitNoSamplesPane_1.add(submitSelectedProblemsRadioButton);
+        
+        selectedProblemsLabel = new JLabel("None selected");
+        selectedProblemsLabel.setToolTipText("None selected");
+        submitNoSamplesPane_1.add(selectedProblemsLabel);
+        
+        selectProblemsButton = new JButton("Select");
+        selectProblemsButton.addActionListener(new ActionListener() {
+            public void actionPerformed(ActionEvent e) {
+                showProblemList();
+            }
+        });
+        selectProblemsButton.setToolTipText("Select Problems");
+        submitNoSamplesPane_1.add(selectProblemsButton);
+        
+        JPanel submitLanguages = new JPanel();
```

Comment:
It would be _much_ clearer if the name of this component reflected its type/usage; that is, if it were named like `submitLanguagesPane` (or similar).  There are numerous places further down in the code where it refers to `submitLangages` and it's not completely clear whether it is talking about a panel, a checkbox, or a button.